src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,106 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_CAPACITY = 10;
+    private int capacity;
+    private int size;
+    private T[] elements;
+
+    public ArrayList() {
+        elements = (T[]) new Object[INITIAL_CAPACITY];
+        capacity = INITIAL_CAPACITY;
+    }
+
     @Override
     public void add(T value) {
-
+        increaseCapacity(capacity * 15);
+        elements[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("index Out Of Bounds");
+        }
+        increaseCapacity(capacity * 15);
+        if (size - index >= 0) {
+            System.arraycopy(elements, index, elements, index + 1,
+                    size - index);
+        }
+        elements[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        increaseCapacity(list.size());
+        for (int i = 0; i < list.size(); i++) {
+            elements[size + i] = list.get(i);
+        }
+        size += list.size();
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < 0 || index > size - 1) {
+            throw new ArrayListIndexOutOfBoundsException("index Out Of Bounds");
+        }
```

Comment:
you have the same logic in several places, let's create a separate method and move this logic there to eliminate duplication